src/Npgsql/NpgsqlBinaryExporter.cs

```diff
@@ -280,19 +280,31 @@ public void Dispose()
         {
             if (_isDisposed) { return; }
 
-            if (!_isConsumed)
+            try
             {
-                // Finish the current CopyData message
-                _buf.Skip(_leftToReadInDataMsg);
-                // Read to the end
-                _connector.SkipUntil(BackendMessageCode.CopyDone);
-                Expect<CommandCompleteMessage>(_connector.ReadMessage());
-                Expect<ReadyForQueryMessage>(_connector.ReadMessage());
+                if (!_isConsumed)
+                {
+                    // Finish the current CopyData message
+                    _buf.Skip(_leftToReadInDataMsg);
+                    try
+                    {
+                        // Read to the end
+                        _connector.SkipUntil(BackendMessageCode.CopyDone);
+                        Expect<CommandCompleteMessage>(_connector.ReadMessage());
+                        Expect<ReadyForQueryMessage>(_connector.ReadMessage());
+                    }
+                    catch (PostgresException e) when (e.SqlState == PostgresErrorCodes.QueryCanceled)
```

Comment:
As a general rule, [Dispose isn't supposed to throw exceptions](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1065?view=vs-2019), so this should swallow and exceptions and log them.

However, QueryCanceled is the one exception that doesn't actually represent an error; if it occurred, this is because the user requested it, and the exception is completely expected. So specifically for that exception, we shouldn't log an error (but rather Debug).